tests/xcm-simulator/src/mocks/parachain.rs

```diff
@@ -515,8 +515,8 @@ impl xcm_executor::Config for XcmConfig {
         FixedRateOfForeignAsset<XcAssetConfig, ShidenXcmFungibleFeeHandler>,
     );
     type ResponseHandler = ();
-    type AssetTrap = ();
-    type AssetClaims = ();
+    type AssetTrap = XcmPallet;
```

Comment:
When you add a pallet instance to a runtime, you need to define its name.

E.g. in case of the relay chain, when `pallet-xcm` is integrated, it's called `XcmPallet`.
In parachain case, it's called a bit differently - `PolkadotXcm`.

This has a significant impact on how the pallet works, even though it seems like a minor change.
Each pallet has some storage items, and when the storage _key_ is constructed, the first thing that comes is the pallet name.

So it's no surprise you didn't see anything trapped, since asset trap functionality was handled by the relay chain's pallet. :see_no_evil: 
I assume this is quite confusing :sweat_smile: , it's normal - I suggest reading this before deep-diving into the code to understand what goes on - [LINK](https://www.shawntabrizi.com/substrate/transparent-keys-in-substrate/).

Reply:
This fixed everything,  I should have been more attentive.
Thanks for help.